libclc/CMakeLists.txt

```diff
@@ -450,10 +450,15 @@ foreach( t ${LIBCLC_TARGETS_TO_BUILD} )
       list(APPEND flags -D__unix__)
     endif()
 
+    set(spirv_flags ${flags})
+    if( ARCH STREQUAL nvptx OR ARCH STREQUAL nvptx64 )
+      list(APPEND spirv_flags -Xclang -fcuda-short-ptr -mllvm -nvptx-short-ptr)
```

Comment:
I *think* this is ok, but you could end up linking the libspirv (with short ptr) to a program compiled without. As builtins don't write pointers, I don't think this is an issue but would be good to test prior to merging and document this.

Reply:
Note I've now reduced the scope of this PR and the option is no longer enabled by default. Thus libclc/libspirv is no longer compiled with this option. If a user passes `-fcuda-short-ptr` they'll see a warning while linking libclc/libspirv which I think is correct to do.

But yes in general we need to consider whether this is okay. I also think it's okay, and I've run several benchmarks with it and not seen a problem. Ideally we'd compile two versions of libspirv for NVPTX, but I don't know if it's going to be worth it for this relatively obscure option.